### PR TITLE
JIT: ARM64 SVE format encodings, `SVE_BH_3A` to `SVE_BH_3B_A`

### PR DESCRIPTION
**Pull request by TIHan**
Contributes to https://github.com/dotnet/runtime/issues/94549

Adds formats
- SVE_BH_3A
- SVE_BH_3B
- SVE_BH_3B_A

Left: Capstone,
Right: Jit
image

### CODE REVIEW

File: src/coreclr/jit/codegenarm64test.cpp [hunk 1]
```diff
@@ -8192,6 +8192,31 @@ void CodeGen::genArm64EmitterUnitTestsSve()
     // IF_SVE_BI_2A
     // MOVPRFX <Zd>, <Zn>
     theEmitter->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, REG_V3, REG_V5);
+
+    // IF_SVE_BH_3A
+    // ADR <Zd>.<T>, [<Zn>.<T>, <Zm>.<T>{, <mod><amount>}]
+    theEmitter->emitIns_R_R_R_I(INS_sve_adr, EA_SCALABLE, REG_V0, REG_V2, REG_V3, 0, INS_OPTS_SCALABLE_S);
```
Comment (a74nh) [inline src/coreclr/jit/codegenarm64test.cpp:8198]: Happy as is, but it'd be nice to have some high registers here to make sure nothing is overflowing.

**Review by a74nh — APPROVED** [review]
LGTM

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -2539,6 +2539,27 @@ void emitter::emitInsSanityCheck(instrDesc* id)
             assert(isVectorRegister(id->idReg2()));
             break;
 
+        case IF_SVE_BH_3A: // .........x.mmmmm ....hhnnnnnddddd -- SVE address generation
+            imm      = emitGetInsSC(id);
+            elemsize = id->idOpSize();
```
Comment (amanasifkhalid) [inline src/coreclr/jit/emitarm64.cpp:2544]: Did you mean to check `elemsize` is `EA_SCALABLE` here?

**Review by amanasifkhalid — APPROVED** [review]
LGTM, too. Thanks!

File: src/coreclr/jit/emitarm64.cpp (hunk 2 above)
Comment (TIHan) [reply src/coreclr/jit/emitarm64.cpp:2544]: Yea I think I did mean to do that.

File: src/coreclr/jit/codegenarm64test.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/codegenarm64test.cpp:8198]: I modified one to test V31 and capstone outputted: `40ACFF04  adr   z0.d, [z2.d, z31.d, lsl #3]` so it all should be good.

**Comment by TIHan** [comment]
@kunalspathak yea, they already implemented it. I'll close this in favor of their work.

**Pull request closed without merging**